subsys/net/ip/net_if.c

```diff
@@ -4998,14 +4998,15 @@ void net_if_init(void)
 
 	STRUCT_SECTION_FOREACH(net_if, iface) {
 
+		init_iface(iface);
```

Comment:
That is certainly possible and we will see it if that happens after this is merged. It is just that the wifi drivers might not work properly (at least the hostap based drivers will not work, I have one cooking right now) if the naming is done before the init as the init will need to mark the interface as "wifi" one. Atm in the upstream code the wifi interface will be Ethernet by default and will get its name set to "eth0" instead of "wlan0". Lets figure out what to do if this change causes issues.